lib/iris/analysis/__init__.py

```diff
@@ -1173,6 +1177,119 @@ def post_process(self, collapsed_cube, data_result, coords, **kwargs):
         return result
 
 
+class Weights(np.ndarray):
```

Comment:
I totally agree with you! No idea why I originally thought it would be a good idea to make this class public.

I made it private now and also removed the special code that handled inherited members in the doc differently for this class.